.github/workflows/validate.yml

```diff
@@ -207,3 +207,77 @@ jobs:
           name: ${{ matrix.package.name }}-${{ matrix.package.shardIndex }}-test-results
           path: packages/**/test-results
           retention-days: 30
+
+  smoke-test:
+    needs: repo
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+          lfs: 'true'
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
+
+      - name: node_modules cache
+        id: npm-cache
+        uses: actions/cache@v4
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-node_modules-${{hashFiles('**/package-lock.json')}}
+          restore-keys: |
+            ${{ runner.os }}-node_modules-
+
+      - name: Install monorepo dependencies (not including smoke-test package)
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        shell: bash
+        env:
+          NODE_OPTIONS: '--max_old_space_size=8192'
+        run: npm ci --prefer-offline --no-audit --progress=false --workspaces --include-workspace-root
+
+      - name: Build and pack iot-app-kit packages
+        run: ./testing/smoke-test/build-pack.sh
+
+      - name: Validate package installation
+        run: npm run validate:installation
+        working-directory: ./testing/smoke-test
+
+      - name: Validate package bundling
+        run: npm run validate:bundling
+        working-directory: ./testing/smoke-test
+
+      - name: Get installed Playwright version
+        id: playwright-version
+        run: echo "::set-output name=version::$(npm info @playwright/test version)"
+        working-directory: ./testing/smoke-test
+
+      - name: Playwright cache (smoke test)
+        uses: actions/cache@v4
+        id: playwright-cache
+        with:
+          path: |
+            ~/.cache/ms-playwright
+          key: '${{ runner.os }}-playwright-smoke-test-${{ steps.playwright-version.outputs.version }}'
+          restore-keys: |
+            ${{ runner.os }}-playwright-smoke-test-
+
+      - name: Install Playwright with dependencies
+        if: steps.playwright-cache.outputs.cache-hit != 'true'
+        run: npx playwright install --with-deps
+        working-directory: ./testing/smoke-test
+
+      - name: Install Playwright's dependencies
+        if: steps.playwright-cache.outputs.cache-hit == 'true'
+        run: npx playwright install-deps
```

Comment:
What is the difference between playwright's deps and with deps? I forget what each of these does.